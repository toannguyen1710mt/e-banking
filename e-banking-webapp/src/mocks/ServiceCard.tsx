import {
  ElectricIcon,
  HouseIcon,
  NetflixIcon,
  PrimeIcon,
  WifiRouterIcon,
} from '@/components';

export const SERVICE_CARD = [
  {
    icon: <NetflixIcon />,
    title: 'Netflix Subscription',
    amount: 100,
  },
  {
    icon: <HouseIcon />,
    title: 'Mortgage',
    amount: 1100,
  },
  {
    icon: <ElectricIcon />,
    title: 'Electric Bill',
    amount: 130,
  },
  {
    icon: <PrimeIcon />,
    title: 'Prime Subscription',
    amount: 90,
  },
  {
    icon: <WifiRouterIcon />,
    title: 'Wifi Subscription',
    amount: 200,
  },
];
