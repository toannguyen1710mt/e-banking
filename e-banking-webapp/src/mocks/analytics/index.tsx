export const ANALYTICS_DATA = [
  {
    title: 'Total Income',
    subtitle: 'Last 30 days income',
    isPositive: true,
    amount: 670000,
    percentageChange: 8,
  },
  {
    title: 'Total Expense',
    subtitle: 'Last 30 days income',
    isPositive: true,
    amount: 270000,
    percentageChange: 8,
  },
  {
    title: 'Total Investment',
    subtitle: 'Last 30 days income',
    isPositive: true,
    amount: 170000,
    percentageChange: 8,
  },
];
