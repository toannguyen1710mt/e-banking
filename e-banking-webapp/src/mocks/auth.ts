// Libs
import { Session } from 'next-auth';

// Interfaces
import { IUser } from '@/interfaces';

export const MOCK_DATA_USER: IUser = {
  accounts: [
    {
      id: 114,
      documentId: 'e5bavd05yrng3g850yazqj0q',
      name: 'OCB',
      accountNumber: '123456789208',
      balance: 3441800,
      type: 'Main',
      currency: '$',
    },
    {
      id: 115,
      documentId: 'kvelrmbr5yan4gyhz4wnoio3',
      name: 'ACB',
      accountNumber: '999332289888',
      balance: 84438800,
      type: 'Checkings',
      currency: '$',
    },
    {
      id: 116,
      documentId: 'nios4sn0bgw5u28nelo8fmao',
      name: 'VID',
      accountNumber: '106868929357',
      balance: 9235800,
      type: 'Savings',
      currency: '$',
    },
  ],
  avatar: 'https://i.ibb.co/SKHPQYq/avatar-default.webp',
  country: 'UK',
  phone: '123456789000',
  postal: '226',
  token:
    'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ODIsImlhdCI6MTczNzA4NzMwNiwiZXhwIjoxNzM5Njc5MzA2fQ.6VPWVxS169pVk-4xBcC9TOj2e1DS10b_l3t9Grs5pU0',
  id: 82,
  documentId: 'ldrhsspx98g1ijlabnrclht8',
  username: 'phong',
  email: 'phong+1@gmail.com',
  provider: 'local',
  confirmed: true,
  blocked: false,
  announcements: true,
  updates: true,
  feedbacksAndSurvey: true,
  events: true,
  generalNotification: true,
  promotions: true,
  eventsNearMe: true,
  createdAt: '2025-01-15T04:54:45.295Z',
  updatedAt: '2025-01-15T04:54:45.623Z',
  publishedAt: '2025-01-15T04:54:45.296Z',
};

export const MOCK_DATA_USER_EMPTY_PROPS: IUser = {
  accounts: [
    {
      documentId: '11',
      accountNumber: '',
      balance: 0,
      currency: '$',
      id: 36,
      name: '',
      type: '',
    },
    {
      documentId: '12',
      accountNumber: '',
      balance: 0,
      currency: '$',
      id: 37,
      name: '',
      type: '',
    },
    {
      documentId: '12',
      accountNumber: '',
      balance: 0,
      currency: '$',
      id: 38,
      name: '',
      type: '',
    },
  ],
  avatar: 'https://i.ibb.co/SKHPQYq/avatar-default.webp',
  country: 'UK',
  email: 'thong+1@gmail.com',
  id: 43,
  phone: '123456789000',
  postal: '226',
  token: '123456',
  username: 'thong',
  documentId: '',
  provider: '',
  confirmed: false,
  blocked: false,
  createdAt: '',
  updatedAt: '',
  publishedAt: '',
};

export const MOCK_SESSION_DATA: Session = {
  user: {
    email: 'phong+1@gmail.com',
    id: 82,
    documentId: 'ldrhsspx98g1ijlabnrclht8',
    username: 'phong',
    provider: 'local',
    confirmed: true,
    blocked: false,
    postal: '344',
    phone: '324345346345',
    country: 'United States',
    avatar: 'https://i.ibb.co/SKHPQYq/avatar-default.webp',
    announcements: true,
    updates: true,
    feedbacksAndSurvey: true,
    events: true,
    generalNotification: true,
    promotions: true,
    eventsNearMe: true,
    createdAt: '2025-01-15T04:54:45.295Z',
    updatedAt: '2025-01-15T04:54:45.623Z',
    publishedAt: '2025-01-15T04:54:45.296Z',
    token:
      'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ODIsImlhdCI6MTczNzA4NzMwNiwiZXhwIjoxNzM5Njc5MzA2fQ.6VPWVxS169pVk-4xBcC9TOj2e1DS10b_l3t9Grs5pU0',
  },
  expires: '2025-01-17T04:24:13.935Z',
};
