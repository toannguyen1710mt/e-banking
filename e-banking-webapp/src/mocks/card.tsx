export const ANALYTICS_CARD = [
  {
    title: 'Total Income',
    isPositive: true,
    subtitle: 'Last 30 days income',
  },
  {
    title: 'Total Expense',
    isPositive: true,
    subtitle: 'Last 30 days expenditure',
  },
  {
    title: 'Total Investment',
    subtitle: 'Last 30 days investment',
  },
];

export const MOCK_LIST_CARD = [
  {
    id: 1,
    documentId: 'zj103n8cbv857otlqpt3ks2h',
    holderName: 'donal trump',
    cardNumber: '123412341234',
    expireAt: '2026-05',
    ccv: '336',
    fullName: null,
    createdAt: '2024-12-18T02:37:12.625Z',
    updatedAt: '2024-12-18T02:37:12.625Z',
    publishedAt: '2024-12-18T02:37:12.623Z',
  },
  {
    id: 2,
    documentId: 'ny4mxu6br8v900c3an9p5u0f',
    holderName: 'Beck Kham',
    cardNumber: '106868929357',
    expireAt: '2028-03',
    ccv: '339',
    fullName: null,
    createdAt: '2024-12-18T04:27:04.764Z',
    updatedAt: '2024-12-18T04:27:04.764Z',
    publishedAt: '2024-12-18T04:27:04.762Z',
  },
  {
    id: 3,
    documentId: 'ljq5kv0f327w5sor2czgs07t',
    holderName: 'Hoang Tran',
    cardNumber: '123412341234',
    expireAt: '2025-07',
    ccv: '333',
    fullName: null,
    createdAt: '2024-12-18T18:24:58.801Z',
    updatedAt: '2024-12-18T18:24:58.801Z',
    publishedAt: '2024-12-18T18:24:58.799Z',
  },
  {
    id: 4,
    documentId: 's0bz93jb4tc5w7wc820o33c2',
    holderName: 'Test Holder Name',
    cardNumber: '123456789876',
    expireAt: '2026-02',
    ccv: '123',
    fullName: null,
    createdAt: '2024-12-19T01:39:29.501Z',
    updatedAt: '2024-12-19T01:39:29.501Z',
    publishedAt: '2024-12-19T01:39:29.498Z',
  },
];
