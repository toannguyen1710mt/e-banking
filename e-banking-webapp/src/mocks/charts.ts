export const MOCK_SPENDING_STATISTIC_CHART_DATA = {
  series: [
    {
      name: 'Data 1',
      data: [10, 41, 35, 51, 49, 91, 133],
    },
    {
      name: 'Data 2',
      data: [28, 69, 73, 102, 123, 142, 153],
    },
  ],
  categories: ['Mon', 'Tue', 'Wed', 'Thur', 'Fri', 'Sat', 'Sun'],
};

export const MOCK_BALANCE_STATISTICS_CHART_DATA = {
  series: [
    {
      name: 'Data 1',
      data: [98, 124, 166, 350, 40, 289, 96],
    },
    {
      name: 'Data 2',
      data: [80, 132, 169, 270, 49, 174, 90],
    },
  ],
  categories: ['Mon', 'Tue', 'Wed', 'Thur', 'Fri', 'Sat', 'Sun'],
};

export const MOCK_SERIES_EXPENSE_ANALYSIS = [22, 45, 35, 40];

export const MASTERCARD_CHART_MOCK = [65, 20, 15];
