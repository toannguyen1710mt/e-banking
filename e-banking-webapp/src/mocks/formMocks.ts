import { AccountType, GlobalType } from '@/interfaces';
import { useForm } from 'react-hook-form';

export const useMockFormGlobal = () => {
  return useForm({
    defaultValues: {
      globalTransfer: {
        fromAccountType: '' as AccountType,
        fromCountryType: '' as GlobalType,
        recipientAccount: '',
        amount: '',
      },
      fromAccountId: '',
      fromCardName: '',
      fromAccountNumber: '',
      fromAccountBalance: 0,
      recipientName: '',
    },
  });
};

export const useMockFormInternal = () => {
  return useForm({
    defaultValues: {
      fromAccountId: '',
      fromCardName: '',
      fromAccountNumber: '',
      fromAccountBalance: 0,
      toAccountId: '',
      internalTransfer: {
        amount: '',
        fromAccountType: '' as AccountType,
        toAccountType: '' as AccountType,
      },
      toCardName: '',
      toAccountNumber: '',
      toAccountBalance: 0,
    },
  });
};
