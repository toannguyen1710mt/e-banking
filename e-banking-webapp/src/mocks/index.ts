export * from './options';
export * from './card';
export * from './charts';
export * from './transaction';
export * from './table';
export * from './transferConfirmData';
export * from './target';
export * from './analytics';
export * from './auth';
export * from './ServiceCard';
export * from './formMocks';
