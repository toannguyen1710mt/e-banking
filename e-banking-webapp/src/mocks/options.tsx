import Link from 'next/link';

// Constants
import { ROUTES } from '@/constants';

// Components
import { SettingIcon, SignOutIcon } from '@/components';

export const MOCK_OPTIONS = [
  {
    key: 'Kenya',
    label: 'Kenya',
  },
  {
    key: 'USA',
    label: 'USA',
  },
];

export const MOCK_CUSTOM_OPTIONS = [
  {
    key: 'Pheroxios Yehudi',
    label: '',
    customOptionElement: (
      <div className='flex flex-col gap-2'>
        <span className='text-xs font-semibold'>Pheroxios Yehudi</span>
        <span className='text-3xs font-semibold text-transparentBlack'>
          Pheroxios@yehudi.com
        </span>
      </div>
    ),
  },
  {
    key: 'Settings',
    label: 'Settings',
    startContent: <SettingIcon />,
    customOptionElement: <Link href={ROUTES.GENERAL}>Settings</Link>,
  },
  {
    key: 'Sign Out',
    label: 'Sign Out',
    startContent: <SignOutIcon />,
  },
];
