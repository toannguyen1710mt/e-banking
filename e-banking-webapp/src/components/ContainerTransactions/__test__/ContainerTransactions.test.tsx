// Libs
import { render } from '@testing-library/react';

// Mocks
import { MOCK_SESSION_DATA } from '@/mocks';

// Components
import { ContainerTransactions } from '../index';

describe('ContainerTransactions Component', () => {
  test('renders with correct data', () => {
    const { container } = render(
      <ContainerTransactions session={MOCK_SESSION_DATA} currentPage={1} />,
    );
    expect(container).toMatchSnapshot();
  });
});
