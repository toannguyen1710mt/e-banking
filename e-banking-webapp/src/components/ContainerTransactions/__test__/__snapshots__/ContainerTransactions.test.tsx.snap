// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContainerTransactions Component renders with correct data 1`] = `<div />`;
